src/server/index.ts

```diff
@@ -54,21 +54,30 @@ export class Server implements IServer {
   private context: ServerContext;
 
   constructor() {
+    const {
+      PORT: port = 3000,
+      ES_SEARCH_HOSTNAME: esHost = 'es-search',
+      ES_SEARCH_PORT: esPort = '9200',
+      MONGO_HOSTNAME: mongoHost = 'mongo',
+      MONGO_PORT: mongoPort = '21017',
+      MONGO_DB_NAME: dbName = 'cse'
+    } = process.env;
+
     this.app = e();
-    this.port = (process.env.PORT as string) || (3000 as number);
+    this.port = port;
     this.configure();
     this.registerMiddleware();
     this.routes();
     this.context = {
       logger,
       tracer: createTracer(),
       searchClient: new SearchClient({
-        host: 'es-search:9200'
+        host: `${esHost}:${esPort}`
       }),
       dbClient: new DbClient({
-        url: 'mongodb://mongo',
-        port: 27017,
-        dbName: 'cse'
+        dbName,
+        url: `mongodb://${mongoHost}`,
+        port: parseInt(mongoPort, 10)
```

Comment:
Can the ports be parsed in a central location instead of every time they're referenced?

Reply:
I tried that at first; the issue that I encountered is that `process.env.MONGO_PORT` is typed `string | undefined`. What I have in place right now is the variable definition handling the `undefined` case,  and therefore `parseInt` always gets a string. If I try to do the `parseInt` closer to the variable definition, it gets slightly messier and requires further testing to be added.